crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1137,6 +1154,20 @@ pub(crate) fn handle_code_action(
         let resolve_data =
             if code_action_resolve_cap { Some((index, params.clone())) } else { None };
         let code_action = to_proto::code_action(&snap, assist, resolve_data)?;
+
+        // Check if the client supports the necessary `ResourceOperation`s.
+        if let Some(changes) = &code_action.edit.as_ref().unwrap().document_changes {
```

Comment:
I missed this `unwrap` in my review, this causes the server to basically panic at all times 😅